src/main/java/com/synopsys/integration/alert/workflow/filter/JsonFieldFilterBuilder.java

```diff
@@ -53,9 +54,19 @@ public JsonFieldFilterBuilder(final JsonExtractor jsonExtractor, final JsonField
             final List<String> contentValues = jsonExtractor.getValuesFromJson(jsonField, notification.getContent());
             logger.debug("Comparing values {} to the configured matcher: {}", contentValues, value);
             for (final String contentValue : contentValues) {
-                if (StringUtils.isNotBlank(contentValue) && (contentValue.equals(value.trim()) || contentValue.matches(value))) {
-                    logger.debug("Match: {}", contentValue);
-                    return true;
+                if (StringUtils.isNotBlank(contentValue)) {
+                    if (contentValue.equals(value.trim())) {
+                        logger.debug("Match: {}", contentValue);
+                        return true;
+                    }
+                    try {
+                        if (contentValue.matches(value)) {
+                            logger.debug("Match: {}", contentValue);
+                            return true;
+                        }
+                    } catch (PatternSyntaxException e) {
```

Comment:
Do we validate the pattern when testing distribution jobs?

Reply:
The problem is, this could either be a Project name OR the Project name pattern from the job configuration. And the issue occurs when you have a project name that has special characters that are not a valid pattern.

Reply:
Can we put a TODO here when we move away from Json Extraction?

Reply:
I am adding a comment and will create a ticket to address this when we move away from Json Extraction